File: JULY-2025/31-07-2025/p898.dart
class Solution {
  int subarrayBitwiseORs(List<int> arr) {
    Set<int> res = <int>{};
    Set<int> cur = <int>{};

    for (var num in arr) {
      Set<int> next = <int>{};
      next.add(num);
      for (var x in cur) {
        next.add(x | num);
      }
      cur = next;
      res.addAll(cur);
    }

    return res.length;
  }
}

File: JULY-2025/31-07-2025/p898.kt
class Solution {
    fun subarrayBitwiseORs(arr: IntArray): Int {
        return arr.fold(mutableSetOf<Int>() to setOf<Int>()) { (all, current), e ->
            val next = (current.map { it or e } + e).toSet()
            all.addAll(next)
            all to next
        }.first.size
    }
}